bodhi/tests/functional/test_updates.py

```diff
@@ -1833,6 +1833,48 @@ def test_obsoletion_unlocked_with_open_stable_request(self, publish, *args):
 
     @mock.patch(**mock_taskotron_results)
     @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.notifications.publish')
+    def test_revoke_action_for_stable_request(self, publish, *args):
+        """Test revoking stable request on testing update"""
+
+        args = self.get_update('bodhi-2.0.0-3.fc17')
+        resp = self.app.post_json('/updates/', args)
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        up.status = UpdateStatus.testing
+        up.request = UpdateRequest.stable
+        self.db.flush()
+
+        resp = self.app.post_json(
+            '/updates/%s/request' % args['builds'],
+            {'request': 'revoke', 'csrf_token': self.get_csrf_token()})
+        eq_(resp.json['update']['request'], None)
+        eq_(resp.json['update']['status'], 'testing')
+        publish.assert_called_with(
+                topic='update.request.revoke', msg=mock.ANY)
+
+    @mock.patch(**mock_taskotron_results)
+    @mock.patch(**mock_valid_requirements)
+    @mock.patch('bodhi.notifications.publish')
+    def test_revoke_action_for_testing_request(self, publish, *args):
+        """Test revoking testing request on pending update"""
+
+        args = self.get_update('bodhi-2.0.0-3.fc17')
+        resp = self.app.post_json('/updates/', args)
+        up = self.db.query(Update).filter_by(title=resp.json['title']).one()
+        up.status = UpdateStatus.pending
+        up.request = UpdateRequest.testing
+        self.db.flush()
+
+        resp = self.app.post_json(
+            '/updates/%s/request' % args['builds'],
+            {'request': 'revoke', 'csrf_token': self.get_csrf_token()})
+        eq_(resp.json['update']['request'], None)
+        eq_(resp.json['update']['status'], 'pending')
```

Comment:
@trishnaguha if the test is failing then I guess there is a bug in the code :/
Since clearly according to your code the status should be ``unpushed``: https://github.com/fedora-infra/bodhi/pull/773/files#diff-55c0d4c740a773e117d1f620f264f793R1119